eis_toolkit/prediction/gradient_boosting.py

```diff
@@ -0,0 +1,215 @@
+from typing import Literal, Optional, Tuple, Union
+
+import numpy as np
+import pandas as pd
+from beartype import beartype
+from sklearn.ensemble import GradientBoostingClassifier, GradientBoostingRegressor
+from sklearn.metrics import classification_report
+from sklearn.model_selection import train_test_split
+
+from eis_toolkit import exceptions
+from eis_toolkit.prediction.model_utils import evaluate_regression_model, tune_model_parameters
+
+
+@beartype
+def gradient_boosting_classifier_train(
+    X: Union[np.ndarray, pd.DataFrame],
```

Comment:
I always find pandas dataframes and series a bit dangerous in comparison to arrays which are usually more simply structured. Probably not worth it at this point to remove functionality but I would have maybe only taken numpy arrays as inputs for the sake of simplicity. 

I also see that dataframe inputs are not tested in the unit tests, I would consider adding a simple test case with a dataframe input instead of an array.

Reply:
On your user side, lots of people don't understand arrays - so having it magically take dataframes/tables and transform them in the background helps some people.

Reply:
If these functions are used as part of a library in interactive Python scripting, you can probably expect the coders to be able to extract arrays from dataframes. You are anyways not passing dataframes or numpy arrays from the QGIS plugin interface as there will anyway be middleware in-between. As I said probably not worth it removing functionality unless some problems actually manifest but simple test cases should be added instead!